File: Web/RMA-Member-Portal/memberportal-app/src/app/shared/models/generic-document.ts
// import { DocumentSetEnum } from 'projects/shared-models-lib/src/lib/enums/document-set.enum';
// import { DocumentStatusEnum } from 'projects/shared-models-lib/src/lib/enums/document-status-enum';
// import { DocumentTypeEnum } from 'projects/shared-models-lib/src/lib/enums/document-type.enum';

import { DocumentSetEnum } from "../enums/document-set.enum";
import { DocumentStatusEnum } from "../enums/document-status-enum";
import { DocumentTypeEnum } from "../enums/document-type.enum";

export class GenericDocument {
  id: number;
  keyName: string;
  keyValue: string;
  systemName: string;
  documentSet: DocumentSetEnum;
  documentType: DocumentTypeEnum;
  required: boolean;
  fileName: string;
  fileExtension: string;
  documentDescription: string;
  documentStatus: DocumentStatusEnum;
  documentUri: string;
  isMemberVisible: boolean;
  isDeleted: boolean;
  createdBy: string;
  createdDate: Date;
  modifiedBy: string;
  modifiedDate: Date;

  fileAsBase64: string;
}

File: Service/ScanCare/RMA.Service.ScanCare.Contracts/Entities/GenericDocument.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;

namespace RMA.Service.ScanCare.Contracts.Entities
{
    public class GenericDocument
    {
        public int Id { get; set; }
        public string KeyName { get; set; }
        public string KeyValue { get; set; }
        public string SystemName { get; set; }
        public DocumentSetEnum DocumentSet { get; set; }
        public DocumentTypeEnum DocumentType { get; set; }
        public bool Required { get; set; }
        public string FileName { get; set; }
        public string FileExtension { get; set; }
        public string DocumentDescription { get; set; }
        public bool IsMemberVisible { get; set; }
        public DocumentStatusEnum DocumentStatus { get; set; }
        public string DocumentUri { get; set; }
        public bool IsDeleted { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime ModifiedDate { get; set; }

        private int? DocumentStatusId
        {
            get => (int?)DocumentStatus;
            set => DocumentStatus = (DocumentStatusEnum)value;
        }

        private int? DocumentSetId
        {
            get => (int?)DocumentSet;
            set => DocumentSet = (DocumentSetEnum)value;
        }

        private int? DocumentTypeId
        {
            get => (int?)DocumentType;
            set => DocumentType = (DocumentTypeEnum)value;
        }
    }
}
